src/symbols.jl

```diff
@@ -94,64 +98,80 @@ function clean_method_path(m::Method)
     return normpath(path)
 end
 
-function cache_methods(f, mod = nothing, exported = false)
+const _global_method_cache = IdDict{Any,Vector{Any}}()
+function methodinfo(@nospecialize(f); types=Tuple, world=typemax(UInt))
```

Comment:
I added this but I think it's possible to do even better. The key observation is that if you go through all functions just once, then you shouldn't need this. As you loop over names in modules, fetch the object and if it's a Callable you can check whether you've processed it before: just keep a running `IdSet` of all previously-processed functions and check `f ∈ processed`. If not, `push!(procssed, f)`, process the whole MethodTable in one go (assigning methods to different modules) and then move on to the next name; if it is already in `processed`, skip to the next name. That will ensure you call `methods(f)` only once for each `f`.

Reply:
This is largely what is being done. Caching of methods should only happen (with a few exceptions) when a function/datatype is encountered from within its parent module. [This](https://github.com/julia-vscode/SymbolServer.jl/pull/175/files#diff-6a3b27e5bfe405fa75cb4196464e0adcR160-R174) then distributes those methods across the different `FunctionStore`s